fdbclient/include/fdbclient/KeyBackedTypes.actor.h

```diff
@@ -540,25 +526,34 @@ class KeyBackedMap {
 	                                 int limit,
 	                                 Snapshot snapshot = Snapshot::False,
 	                                 Reverse reverse = Reverse::False) const {
-		Key beginKey = begin.present() ? packKey(begin.get()) : subspace.begin;
-		Key endKey = end.present() ? packKey(end.get()) : subspace.end;
 
-		typename transaction_future_type<Transaction, RangeResult>::type getRangeFuture =
-		    tr->getRange(KeyRangeRef(beginKey, endKey), GetRangeLimits(limit), snapshot, reverse);
+		if constexpr (is_transaction_creator<Transaction>) {
+			return runTransaction(tr, [=, self = *this](decltype(tr->createTransaction()) tr) {
```

Comment:
Overall the changes to key-backed types look good to me. I have just one small nit, which is that we are shadowing `tr` with the parameter to our lambda function, which could lead to confusion.

I also suspect that to generalize this more, we could push this decision into the `runTransaction` function and make it easy to support different types across all of our function in `KeyBackedTypes`. That doesn't need to be done here, though.